### PR TITLE
Add windows 11 template

### PR DESCRIPTION
**Pull request by fossedihelm**
Signed-off-by: fossedihelm <ffossemo@redhat.com>

**What this PR does / why we need it**:
Add windows 11 template. Windows 11 information currently is not present in osinfo-db (there is an open PR [https://gitlab.com/libosinfo/osinfo-db/-/merge_requests/362](https://gitlab.com/libosinfo/osinfo-db/-/merge_requests/362)); fot this reason has been added an entry fot it in [https://github.com/kubevirt/common-templates/tree/master/osinfo-db-override/os](https://github.com/kubevirt/common-templates/tree/master/osinfo-db-override/os)
This PR requires [https://github.com/kubevirt/kubevirt/pull/7103](https://github.com/kubevirt/kubevirt/pull/7103) since the latter introduce the TPM 2.0 support, required by Windows 11

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add windows 11 template
```

### CODE REVIEW

File: templates/windows11.tpl.yaml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      app: ${NAME}
+    annotations:
+      vm.kubevirt.io/validations: |
+        [
+          {
+            "name": "minimal-required-memory",
+            "path": "jsonpath::.spec.domain.resources.requests.memory",
+            "rule": "integer",
+            "message": "This VM requires more memory.",
+            "min": 4294967296
```
Comment (ksimon1) [inline templates/windows11.tpl.yaml:53]: Can you please create a new record in this folder https://github.com/kubevirt/common-templates/tree/master/osinfo-db-override/os, which will stand as a "replacement" for missing record in osinfo-db (e.g. copy it from osinfo PR).

**Comment by ksimon1** [comment]
Please create new test lane here: https://github.com/openshift/release/blob/master/ci-operator/config/kubevirt/common-templates/kubevirt-common-templates-master.yaml#L155. You will have to create a new windows 11 image for this test lane (e.g. create it in your private quay repo, and I will push it to testing repo).

File: templates/windows11.tpl.yaml (hunk 1 above)
Comment (fossedihelm) [reply templates/windows11.tpl.yaml:53]: Oh, didn't know about that. Thanks so much the info. will adjust it.

**Comment by fossedihelm** [comment]
> Please create new test lane here: https://github.com/openshift/release/blob/master/ci-operator/config/kubevirt/common-templates/kubevirt-common-templates-master.yaml#L155. You will have to create a new windows 11 image for this test lane (e.g. create it in your private quay repo, and I will push it to testing repo).

Thanks, I updated the commit and the description. I will send you this as soon as possible

File: generate-templates.yaml [hunk 2]
```diff
@@ -374,3 +374,15 @@
...
     vars:
       osinfoname: win10
+
+  - name: Generate Windows 11 templates
+    template:
+      src: windows11.tpl.yaml
+      dest: "{{ playbook_dir }}/dist/templates/windows11-{{ item.workload }}-{{ item.flavor }}.yaml"
+    with_items:
+    - {flavor: medium, workload: desktop,         memsize: "4Gi", cpus: 1, iothreads: False, emulatorthread: False, multiqueue: False, tablet: True, default: True}
+    - {flavor: medium, workload: highperformance, memsize: "4Gi", cpus: 1, iothreads: True,  emulatorthread: True,  multiqueue: True,  tablet: True, default: False}
```
Comment (dominikholler) [inline generate-templates.yaml:384]: Windows 11 requires at least 2 cores: https://docs.microsoft.com/en-us/windows/whats-new/windows-11-requirements . Do we want to support singe core?

**Review by dominikholler — CHANGES_REQUESTED** [review]

File: generate-templates.yaml (hunk 2 above)
Comment (dominikholler) [reply generate-templates.yaml:384]: I see, the template uses two cores, and this is just about the cpus.
@ksimon1 Should we align windows 10, too?

Comment (fossedihelm) [reply generate-templates.yaml:384]: Thanks for the review! Maybe I am confusing something but here I put the number of cpu. I put the minimum requirement of cores here
[https://github.com/kubevirt/common-templates/blob/e56147bc86cfdeffa638898ca15dea2a07fffc22/templates/windows11.tpl.yaml#L117](https://github.com/kubevirt/common-templates/blob/e56147bc86cfdeffa638898ca15dea2a07fffc22/templates/windows11.tpl.yaml#L117)
Am I missing something?

Comment (fossedihelm) [reply generate-templates.yaml:384]: Sorry I read now the second comment :(

**Comment by fossedihelm** [comment]
@ksimon1 I pushed changes in `test-windows.sh`. Since windows 11 doesn't let me to use `Administrator` as username I made it a variable in the script. Here's the win11 image [https://quay.io/repository/ffossemo/windows11](https://quay.io/repository/ffossemo/windows11). I'm going to create the windows 11 lane. Thanks

**Comment by ksimon1** [comment]
> @ksimon1 I pushed changes in `test-windows.sh`. Since windows 11 doesn't let me to use `Administrator` as username I made it a variable in the script. Here's the win11 image https://quay.io/repository/ffossemo/windows11. I'm going to create the windows 11 lane. Thanks

The windows 11 image is available on quay in testing repo. Now just create new test lane with this string inside command attribute
`export TARGET=windows11 && make e2e-tests`

**Comment by fossedihelm** [comment]
Add validation for the minimum number of cores

File: generate-templates.yaml (hunk 2 above)
Comment (dominikholler) [reply generate-templates.yaml:384]: ok, thanks, let's keep it as proposed, and check how the new UI is going to handle it.
cc: @yaacov

Comment (yaacov) [reply generate-templates.yaml:384]: hi, sorry for the layman quastion
cpus and threads are per socket/cpu or per machine ?

**Comment by fossedihelm** [comment]
Windows11 lane succeed https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-kubevirt-common-templates-master-e2e-azure-common-templates-windows11. 
Rollback temporary commit and add a fix to a typo.

**Comment by fossedihelm** [comment]
@ksimon1 @dominikholler 
Thinking about this: https://github.com/kubevirt/common-templates/pull/426#issuecomment-1072459365

https://github.com/kubevirt/common-templates/blob/dfe0add8c4ad94b66a0c2780c0cd4e2a80d61c30/templates/windows11.tpl.yaml#L77
I am not sure about this change, since win 11 requires at least 1 cpu with 2 cores, but I suppose that 2 cpu with 1 core is allowed. AFAIK there isn't a way to make a sort of conditional validation. Any suggestions? Thanks

**Review by dominikholler — APPROVED** [review]
/lgtm

**Pull request merged**